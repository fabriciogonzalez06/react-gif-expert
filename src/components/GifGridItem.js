import React from 'react'
import PropTypes from 'prop-types';

function GifGridItem({ id, title, url }) {
    return (
        <div className="card1 animate__animated animate__bounce" >

            <img src={url} alt={title} />
            <p> {title} </p>
          
        </div>

    )
}

GifGridItem.propTypes = {
    title:PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    
}

export default GifGridItem;
