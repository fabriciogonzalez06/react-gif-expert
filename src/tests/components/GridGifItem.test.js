import React from 'react'
import GifGridItem from '../../components/GifGridItem';
import {shallow} from 'enzyme';

describe('Pruebas en <GifGridItem/>', () => {
    
    const url='abc.com';
    const title='Avatar';

    const wrapper= shallow(<GifGridItem url={url} title={title} />);
    test('DeberÃ­a mostrar el componente correctamente', () => {
        
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img= wrapper.find('img');

        //propio de enzyme prorpos
        // console.log(img.prop());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });


    test('debe tener animate_bounce', () => {

        const div= wrapper.find('div');
        expect(div.prop('className').includes('animate__bounce')).toBe(true);
    })
    
    
    
    

})
