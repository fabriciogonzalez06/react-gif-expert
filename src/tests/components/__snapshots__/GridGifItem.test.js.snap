// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Deber√≠a mostrar el componente correctamente 1`] = `
<div
  className="card1 animate__animated animate__bounce"
>
  <img
    alt="Avatar"
    src="abc.com"
  />
  <p>
     
    Avatar
     
  </p>
</div>
`;
